src/player/Player.php

```diff
@@ -1618,6 +1621,45 @@ public function pickBlock(Vector3 $pos, bool $addTileNBT) : bool{
 		return true;
 	}
 
+	public function pickEntity(int $entityId, bool $addEntityNBT): bool{
+		$entity = $this->getWorld()->getEntity($entityId);
+		if($entity === null){
+			return true;
+		}
+
+		$item = $entity->getPickedItem($addEntityNBT);
+
+		$ev = new PlayerEntityPickEvent($this, $entity, $item);
+		$existingSlot = $this->inventory->first($item);
+		if($existingSlot === -1 && ($this->hasFiniteResources() || $this->isSpectator())){
+			$ev->cancel();
+		}
+		$ev->call();
+
+		if(!$ev->isCancelled()){
+			if($existingSlot !== -1){
+				if($existingSlot < $this->inventory->getHotbarSize()){
+					$this->inventory->setHeldItemIndex($existingSlot);
+				}else{
+					$this->inventory->swap($this->inventory->getHeldItemIndex(), $existingSlot);
+				}
+			}else{
+				$firstEmpty = $this->inventory->firstEmpty();
+				if($firstEmpty === -1){ //full inventory
+					$this->inventory->setItemInHand($item);
+				}elseif($firstEmpty < $this->inventory->getHotbarSize()){
+					$this->inventory->setItem($firstEmpty, $item);
+					$this->inventory->setHeldItemIndex($firstEmpty);
+				}else{
+					$this->inventory->swap($this->inventory->getHeldItemIndex(), $firstEmpty);
+					$this->inventory->setItemInHand($item);
+				}
+			}
+		}
```

Comment:
If you want to reuse this logic, extract it from `pickBlock` into a reusable method. Don't duplicate the code.